src/lib/mina.ts

```diff
@@ -86,33 +79,45 @@ function createTransaction(
   f: () => unknown,
   { fetchMode = 'cached' as FetchMode, isFinalRunOutsideCircuit = true } = {}
 ): Transaction {
-  if (currentTransaction !== undefined) {
+  if (currentTransaction.has()) {
     throw new Error('Cannot start new transaction within another transaction');
   }
   let feePayerKey =
     feePayer instanceof PrivateKey ? feePayer : feePayer?.feePayerKey;
   let fee = feePayer instanceof PrivateKey ? undefined : feePayer?.fee;
   let memo = feePayer instanceof PrivateKey ? '' : feePayer?.memo ?? '';
 
-  currentTransaction = {
+  let transactionId = currentTransaction.enter({
     sender: feePayerKey,
     parties: [],
     nextPartyIndex: 0,
     fetchMode,
     isFinalRunOutsideCircuit,
-  };
+  });
 
   try {
     // run circuit
-    Circuit.runAndCheck(f);
+    let err: any;
+    while (true) {
+      // this is such a ridiculous hack
```

Comment:
lol what's happening here? Can you add an explanation in a comment?

Reply:
explained here: https://github.com/o1-labs/snarkyjs/pull/280#discussion_r924160029
Ok will add a better comment!